fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java

```diff
@@ -349,7 +350,14 @@ public String toString() {
             double val = (double) Optional.ofNullable(value).orElse((double) 0);
             BigDecimal decimal = BigDecimal.valueOf(val);
             return decimal.stripTrailingZeros().toPlainString();
-        } else if (type.isDecimalV2()) {
+        }
+
+        if (ConnectContext.get() != null &&
+                !ConnectContext.get().getSessionVariable().isCboDecimalCastStringStrict()) {
+            return String.valueOf(value);
```

Comment:
Is this just for clients could turn off if they want compatible with old version?

Reply:
yeah，you are right，just for compatible